library/block_cipher.c

```diff
@@ -42,26 +82,38 @@ void mbedtls_block_cipher_free(mbedtls_block_cipher_context_t *ctx)
 int mbedtls_block_cipher_setup(mbedtls_block_cipher_context_t *ctx,
                                mbedtls_cipher_id_t cipher_id)
 {
-    switch (cipher_id) {
+    ctx->id = (cipher_id == MBEDTLS_CIPHER_ID_AES) ? MBEDTLS_BLOCK_CIPHER_ID_AES :
+              (cipher_id == MBEDTLS_CIPHER_ID_ARIA) ? MBEDTLS_BLOCK_CIPHER_ID_ARIA :
+              (cipher_id == MBEDTLS_CIPHER_ID_CAMELLIA) ? MBEDTLS_BLOCK_CIPHER_ID_CAMELLIA :
+              MBEDTLS_BLOCK_CIPHER_ID_NONE;
+
+#if defined(MBEDTLS_BLOCK_CIPHER_SOME_PSA)
+    if (psa_can_do_cipher(cipher_id) &&
```

Comment:
This is wrong, and it only compiles because C's type system is so weak: you're passing a `mbedtls_cipher_id_t` where (currently) a `psa_algorithm_t` is expected. I think we want to first get the PSA key type, then call this with `key_type, ALG_ECB_NO_PADDING` (see my comment about the signature of this function).